src/nvim/eval/funcs.c

```diff
@@ -7494,17 +7494,17 @@ static void f_readfile(typval_T *argvars, typval_T *rettv, FunPtr fptr)
          * small, to avoid repeatedly 'allocing' large and
          * 'reallocing' small. */
         if (prevsize == 0) {
-          prevsize = (long)(p - start);
+          prevsize = (long)(p - start);  // NOLINT(google-readability-casting)
```

Comment:
urgl, this is also a (pattern of) valid cast which shouldn't be an error.. (though maybe the real issue is that we should get rid of `long` eventually, it is an useless type which is size of ptr everywhere except for 64-bit MSVC)

Reply:
> (though maybe the real issue is that we should get rid of long eventually, it is an useless type which is size of ptr everywhere except for 64-bit MSVC)

And Linux x32

Reply:
Would it be too janky to have a macro that only casts when it needs to? Something like (pseudocode):

```c
#ifdef WIN32
  #define CAST_TO_LONG (long)
#else
  #define CAST_TO_LONG
#endif
```

Alternatively I'm fine with just reverting this and the ones similar to this one, there are so few of them that it's not a big deal IMO.

Reply:
that's.. even worse. If just reverting this is okay, that would be better :]